### PR TITLE
Script Action new numeration

### PR DESCRIPTION
**Pull request by FS-21**
TO-DO:
Update documentation & What's new.

### CODE REVIEW

**Comment by FS-21** [comment]
imagen
// 10000-10999 are team (aka ingame) actions					
Attack	RepeatAttackCloser	develop	76	10000	x
Attack	SingleAttackCloser	develop	80	10001	x
Attack	RepeatAttackTypeCloser	develop	86	10002	x
Attack	SingleAttackTypeCloser	develop	90	10003	x
Attack	RandomAttackTypeCloser	develop	104	10004	x
Attack	RepeatAttackFarther	develop	77	10005	x
Attack	SingleAttackFarther	develop	81	10006	x
Attack	RepeatAttackTypeFarther	develop	87	10007	x
Attack	SingleAttackTypeFarther	develop	91	10008	x
Attack	RandomAttackTypeFarther	develop	105	10009	x
A+A16:E45ttack	RepeatAttackCloserThreat	develop	74	10010	x
Attack	SingleAttackCloserThreat	develop	78	10011	x
Attack	RepeatAttackTypeCloserThreat	develop	84	10012	x
Attack	SingleAttackTypeCloserThreat	develop	88	10013	x
Attack	RepeatAttackFartherThreat	develop	75	10014	x
Attack	SingleAttackFartherThreat	develop	79	10015	x
Attack	RepeatAttackTypeFartherThreat	develop	85	10016	x
Attack	SingleAttackTypeFartherThreat	develop	89	10017	x
Move	MoveToEnemyCloser	develop	95	10100	x
Move	MoveToTypeEnemyCloser	develop	99	10101	x
Move	RandomMoveToTypeEnemyCloser	develop	106	10102	x
Move	MoveToFriendlyCloser	develop	97	10103	x
Move	MoveToTypeFriendlyCloser	develop	101	10104	x
Move	RandomMoveToTypeFriendlyCloser	develop	108	10105	x
Move	MoveToEnemyFarther	develop	96	10106	x
Move	MoveToTypeEnemyFarther	develop	100	10107	x
Move	RandomMoveToTypeEnemyFarther	develop	107	10108	x
Move	MoveToFriendlyFarther	develop	98	10109	x
Move	MoveToTypeFriendlyFarther	develop	102	10110	x
Move	RandomMoveToTypeFriendlyFarther	develop	109	10111	x

// 12000-12999 are suplementary/setup pre-actions
Attack-Move configuration	WaitIfNoTarget	develop	92	12000	x
Move configuration	ModifyTargetDistance	develop	103	12004	x
Move configuration	SetMoveMissionEndMode	develop	110	12005	x

// 14000-14999 are utility actions (angernodes manipulation)

//16000-16999 are flow control actions (jumps, change script, repeat, etc)
Jump	RandomSkipNextAction	develop	113	16005	x
Jump	StopForceJumpCountdown	develop	124	16004	x
Jump	NextLineForceJumpCountdown	develop	125	16002	x
Jump	SameLineForceJumpCountdown	develop	126	16003	x
Trigger manipulation	DecreaseCurrentAITriggerWeight	develop	82	16103	x
Trigger manipulation	IncreaseCurrentAITriggerWeight	develop	83	16102	x
Trigger manipulation	TeamWeightReward	develop	93	16101	x
Trigger manipulation	PickRandomScript	develop	94	16100	x				
Triggers manipulation	UnregisterGreatSuccess	develop	111	16109	x

// 18000-18999 are variable actions
LocalVariableSet			500	18000	x
LocalVariableAdd			501	18001	x
LocalVariableMinus			502	18002	x
LocalVariableMultiply			503	18003	x
LocalVariableDivide			504	18004	x
LocalVariableMod			505	18005	x
LocalVariableLeftShift			506	18006	x
LocalVariableRightShift			507	18007	x
LocalVariableReverse			508	18008	x
LocalVariableXor			509	18009	x
LocalVariableOr			510	18010	x
LocalVariableAnd			511	18011	x
GlobalVariableSet			512	18012	x
GlobalVariableAdd			513	18013	x
GlobalVariableMinus			514	18014	x
GlobalVariableMultiply			515	18015	x
GlobalVariableDivide			516	18016	x
GlobalVariableMod			517	18017	x
GlobalVariableLeftShift			518	18018	x
GlobalVariableRightShift			519	18019	x
GlobalVariableReverse			520	18020	x
GlobalVariableXor			521	18021	x
GlobalVariableOr			522	18022	x
GlobalVariableAnd			523	18023	x
LocalVariableSetByLocal			524	18024	x
LocalVariableAddByLocal			525	18025	x
LocalVariableMinusByLocal			526	18026	x
LocalVariableMultiplyByLocal			527	18027	x
LocalVariableDivideByLocal			528	18028	x
LocalVariableModByLocal			529	18029	x
LocalVariableLeftShiftByLocal			530	18030	x
LocalVariableRightShiftByLocal			531	18031	x
LocalVariableReverseByLocal			532	18032	x
LocalVariableXorByLocal			533	18033	x
LocalVariableOrByLocal			534	18034	x
LocalVariableAndByLocal			535	18035	x
GlobalVariableSetByLocal			536	18036	x
GlobalVariableAddByLocal			537	18037	x
GlobalVariableMinusByLocal			538	18038	x
GlobalVariableMultiplyByLocal			539	18039	x
GlobalVariableDivideByLocal			540	18040	x
GlobalVariableModByLocal			541	18041	x
GlobalVariableLeftShiftByLocal			542	18042	x
GlobalVariableRightShiftByLocal			543	18043	x
GlobalVariableReverseByLocal			544	18044	x
GlobalVariableXorByLocal			545	18045	x
GlobalVariableOrByLocal			546	18046	x
GlobalVariableAndByLocal			547	18047	x
LocalVariableSetByGlobal			548	18048	x
LocalVariableAddByGlobal			549	18049	x
LocalVariableMinusByGlobal			550	18050	x
LocalVariableMultiplyByGlobal			551	18051	x
LocalVariableDivideByGlobal			552	18052	x
LocalVariableModByGlobal			553	18053	x
LocalVariableLeftShiftByGlobal			554	18054	x
LocalVariableRightShiftByGlobal			555	18055	x
LocalVariableReverseByGlobal			556	18056	x
LocalVariableXorByGlobal			557	18057	x
LocalVariableOrByGlobal			558	18058	x
LocalVariableAndByGlobal			559	18059	x
GlobalVariableSetByGlobal			560	18060	x
GlobalVariableAddByGlobal			561	18061	x
GlobalVariableMinusByGlobal			562	18062	x
GlobalVariableMultiplyByGlobal			563	18063	x
GlobalVariableDivideByGlobal			564	18064	x
GlobalVariableModByGlobal			565	18065	x
GlobalVariableLeftShiftByGlobal			566	18066	x
GlobalVariableRightShiftByGlobal			567	18067	x
GlobalVariableReverseByGlobal			568	18068	x
GlobalVariableXorByGlobal			569	18069	x
GlobalVariableOrByGlobal			570	18070	x
GlobalVariableAndByGlobal			571	18071	x

//19000-19999 are miscellanous/uncategorized actions					
Area Guard	TimedAreaGuard	develop	71	19000	x
Transport load	LoadIntoTransports	develop	72	19005	x
Aircraft ammo	WaitUntilFullAmmo	develop	73	19001	x
Follow	GatherAroundLeader	develop	112	19002	x

File: src/Ext/Script/Body.h [hunk 1]
```diff
@@ -19,126 +19,136 @@
...
+	RandomMoveToTypeEnemyCloser = 10102,
+	MoveToFriendlyCloser = 10103,
+	MoveToTypeFriendlyCloser = 10104,
+	RandomMoveToTypeFriendlyCloser = 10105,
+	MoveToEnemyFarther = 10106,
+	MoveToTypeEnemyFarther = 10107,
+	RandomMoveToTypeEnemyFarther = 10108,
+	MoveToFriendlyFarther = 10109,
+	MoveToTypeFriendlyFarther = 10110,
+	RandomMoveToTypeFriendlyFarther = 10111,
```
Comment (Metadorius) [inline src/Ext/Script/Body.h:52]: Didn't we agree on 50 numbers ranges instead of 100?

File: src/Ext/Script/Body.h [hunk 2]
```diff
@@ -19,126 +19,136 @@
...
+	SetMoveMissionEndMode = 12005,
+
+	// 14000-14999 are utility actions (angernodes manipulation)
+
+
+	//16000-16999 are flow control actions (jumps, change script, repeat, etc)
+	RandomSkipNextAction = 16005,
+	StopForceJumpCountdown = 16004,
+	NextLineForceJumpCountdown = 16002,
+	SameLineForceJumpCountdown = 16003,
```
Comment (Metadorius) [inline src/Ext/Script/Body.h:66]: Why are those actions not in order?

File: src/Ext/Script/Body.h [hunk 3]
```diff
@@ -19,126 +19,136 @@
...
+	DecreaseCurrentAITriggerWeight = 16103,
+	IncreaseCurrentAITriggerWeight = 16102,
+	TeamWeightReward = 16101,
```
Comment (Metadorius) [inline src/Ext/Script/Body.h:69]: Flow control means jumps, loops, in other words - changing the execution flow of the script. This has nothing to do with flow control, this is what utility actions range is for. Also still reversed numbers?

File: src/Ext/Script/Body.h [hunk 4]
```diff
@@ -19,126 +19,136 @@
...
+	PickRandomScript = 16100,
+	UnregisterGreatSuccess = 16109,
```
Comment (Metadorius) [inline src/Ext/Script/Body.h:71]: Same for this one. Utility action.

**Review by Metadorius — CHANGES_REQUESTED** [review]

**Comment by FS-21** [comment]
Applied feedback, if all is fine I'll start with the documentation changes.

**Comment by Metadorius** [comment]
Will review in a few days, feeling not good a bit atm.

**Review by Metadorius — COMMENTED** [review]
Looks good so far. Fix that lint check though.

**Comment by Metadorius** [comment]
Also I'd make subcategories in docs for each range.

File: docs/Whats-New.md [hunk 5]
```diff
@@ -142,81 +142,81 @@ You can use the migration utility (can be found on [Phobos supplementaries repo]
...
-  565=Global variable mod by global variable,27,0,1,[LONG DESC]
-  566=Global variable leftshift by global variable,27,0,1,[LONG DESC]
-  567=Global variable rightshift by global variable,27,0,1,[LONG DESC]
-  568=Global variable reverse by global variable,27,0,1,[LONG DESC]
-  569=Global variable xor by global variable,27,0,1,[LONG DESC]
-  570=Global variable or by global variable,27,0,1,[LONG DESC]
-  571=Global variable and by global variable,27,0,1,[LONG DESC]
+  10100=Timed Area Guard,20,0,1,[LONG DESC]
+  10103=Load Onto Transports,0,0,1,[LONG DESC]
+  10101=Wait until ammo is full,0,0,1,[LONG DESC]
```
Comment (Metadorius) [inline docs/Whats-New.md:147]: The rest needs to be done too. Maybe not in this PR but still.

Comment (FS-21) [reply docs/Whats-New.md:147]: I have very little knowledge in FA2 map editor INI tags. Maybe secsome or another collaborator could help.

**Pull request merged**